### PR TITLE
Add Jupyter Backends

### PR DESCRIPTION
**Pull request by akaszynski**
tl;dr
Jupyter + `pyvista` = Awesome

- Interactive webpages with `sphinx-jupyter`
- Great plots without `xvfb` using `ipygany`
- Use `pyvista.Plotter` class rather than learning a new interface for each backend.
- Improve documentation for jupyter

#### Details

Propose that we support a variety of backends when plotting within a Jupyter notebook:

- Server-side rendering with PyVista streaming to the notebook through
  [ipyvtk-simple](https://github.com/Kitware/ipyvtk-simple)
- Client-side rendering with [ipygany](https://github.com/QuantStack/ipygany) using ``threejs``.
- Client-side rendering using [panel](https://github.com/holoviz/panel) using ``vtk.js``.
- Client-side rendering with [itkwidgets](https://github.com/InsightSoftwareConsortium/itkwidgets) using ``itk.js`` and ``vtk.js``.
- Static images.

https://user-images.githubusercontent.com/11981631/112939384-9ade3b80-90e8-11eb-82f2-ba7008ba5270.mp4

Sample Code:
```py
import pyvista as pv
from pyvista import examples

mesh = examples.download_bunny()
    
pl = pv.Plotter()
pl.add_mesh(mesh, color='lightgrey')
pl.background_color = 'white'
pl.camera_position = 'xy'
pl.show(jupyter_backend='ipygany')
```

#### Bonus
Move Mac OS unit testing into GitHub actions.  Reduces unit test times as we can run parallel tests spread out over Azure and GitHub Actions. Plus, Mac OS seems to have better support on GitHub actions (e.g. Python is easier to install).

### CODE REVIEW

**Comment by akaszynski** [comment]
@adeak, it's a WIP, but it's ready for you to start bleeding over it.  It needs a critical eye.

**Comment by adeak** [comment]
I'll try to find time to look at it, but it's huge and I'm not familiar at all with the notebook ecosystem, so again no promises that I'll be useful.

**Comment by akaszynski** [comment]
@pyvista/developers, this is ready for review.
